mantle/kola/tests/coretest/core.go

```diff
@@ -378,3 +382,76 @@ func TestMachineID() error {
 	}
 	return nil
 }
+
+// TestRHCOSGrowpart tests whether rhcos-growpart was run successfully by checking
+// whether rhcos-growpart.service exited normally and root partition is >= 8 GB.
+func TestRHCOSGrowpart() error {
+	// check that rhcos-growpart.service was run and exited normally
+	// value of MESSAGE_ID= is the SD_MESSAGE_UNIT_STARTED macro's value from
+	// https://github.com/systemd/systemd/blob/master/src/systemd/sd-messages.h
+	c := exec.Command("journalctl", "-o", "json", "MESSAGE_ID=39f53479d3a045ac8e11786248231fbf",
+		"UNIT=rhcos-growpart.service")
+	out, err := c.Output()
+	if err != nil {
+		return fmt.Errorf("Err: %s\n Out: %v", err, out)
+	}
+
+	var journalOutput map[string]string
+	if err := json.Unmarshal(out, &journalOutput); err != nil {
+		return fmt.Errorf("Error getting journalclt output for rhcos-growpart.service: %s", err)
+	}
+	if journalOutput["JOB_RESULT"] != "done" {
+		return fmt.Errorf("rhcos-growpart.service did not run successfully\n Journalctl output: %q", out)
+	}
```

Comment:
Can we factor this logic out into a separate function to share across both functions?